src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -2580,7 +2581,119 @@ class TypeSpecTest {
     }
   }
 
+  @Test fun basicDelegateTest() {
+    val type = TypeSpec.classBuilder("Guac")
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addParameter("somethingElse", String::class)
+            .build())
+        .addSuperinterface(ParameterizedTypeName.get(Consumer::class, String::class), CodeBlock.of("{ println(it) }"))
+        .build()
+
+    val expect = """
+        |package com.squareup.tacos
+        |
+        |import java.util.function.Consumer
+        |import kotlin.String
+        |
+        |class Guac(somethingElse: String) : Consumer<String> by { println(it) }
```

Comment:
This works? That's crazy!

Reply:
turns out it doesn't, only this does `by Consumer({ println(it) })`

Reply:
Fix the test?